### PR TITLE
Draft of contributing guide v2

### PR DESCRIPTION
**Pull request by Half-Shot**
This is my attempt at trying to be a bit more verbose about our contrib guidelines for bridges. The aim here is to have this be linked to by our other projects.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: Consider clarifying what the target of the version is (i.e. "which version _of the bridge_ you are using")

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: Consider adding a bad example to counter this.

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: Even though there are only 3 of them, should the labels be formatted as a (sub)list?

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: maintainers -> maintainers'

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: The last sentence ends abruptly; did it accidentally a few words?

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: Maybe change "fluff" to "going off-topic" to reduce idiomatic language for non-native English speakers.

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: respositories -> repository's (unless it was meant to be plural?)

Also, if "this" refers to the features being written, maybe change "remember to document this" to "remember to document them"

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: To mesh better with the non-imperative tense of the points below, maybe use "please ensure it [the PR] follows these best practises".

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: prompty -> promptly

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: Maybe this should explain how to bootstrap getting the PR number for the changelog file, like [synapse's contributing guide](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#how-do-i-know-what-to-call-the-changelog-file-before-i-create-the-pr) does.

Also, this bullet isn't meant to be a sub-entry of the previous bullet, right?

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md:92]: If a repo doesn't have a `develop` branch, should we recommend targeting `main`, or tell people to defer to the contribution guide of the repo they're working with?

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: This should clarify whether this follows Synapse's policy of requiring legally-identifiable names. I assume it does.

**Review by AndrewFerr — CHANGES_REQUESTED** [review]
This is a very useful guide--I'm already using it :smiley: 

Just made a few comments & requests.

File: CONTRIBUTING.md
Comment (Half-Shot) [reply CONTRIBUTING.md]: Ironic really. I think what I was trying to say is "use plain english", so now I've said that.

File: CONTRIBUTING.md
Comment (Half-Shot) [reply CONTRIBUTING.md:92]: We should probably head towards using either `develop` or `main` as the standard branch across all projects, but in the meantime let's call out divergences in the contributing file (I think it might just be hookshot, anyway).

File: CONTRIBUTING.md
Comment (Half-Shot) [reply CONTRIBUTING.md]: Yes, have made clear that we follow all the rules in the sign off section.

File: CONTRIBUTING.md
Comment (Half-Shot) [reply CONTRIBUTING.md]: Have fixed the bullets.

Also, have linked back to Synapse more, and made a point of mentioning their guide on PR numbers.

File: CONTRIBUTING.md
Comment (thibaultamartin) [inline CONTRIBUTING.md]:
```suggestion
```

Do we have actual plans for a multi-lingual documentation system? This requires significant additional work

File: CONTRIBUTING.md
Comment (thibaultamartin) [inline CONTRIBUTING.md]: > please reach out to us

What is the best way to reach out to maintainers of each project (PR comment, message in a Matrix room, other)? Is the preferred way universal across matrix-org or is it per-project (e.g. one project will prefer PR comments, another will prefer conversations in a matrix room)?

File: CONTRIBUTING.md
Comment (Half-Shot) [reply CONTRIBUTING.md]: Good point. Maybe our contrib file shouldn't promise what isn't  happening soon. We should at least state that it's in english though. (Or should we merge with the line about plain english....yes lets).

File: CONTRIBUTING.md
Comment (Half-Shot) [reply CONTRIBUTING.md]: I think we'll probably mention this in each specific contributing file.

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: amonst -> amongst (ugh, sorry I missed this the first time...)

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: Nitpick: maybe add a hyphen for "Matrix-based"?

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: it's entirity -> its entirety

**Review by AndrewFerr — CHANGES_REQUESTED** [review]
Some final minor points

File: CONTRIBUTING.md
Comment (AndrewFerr) [inline CONTRIBUTING.md]: it's -> its

**Review by AndrewFerr — CHANGES_REQUESTED** [review]
Just one last nitpick, then I'm satisfied :)

**Review by AndrewFerr — APPROVED** [review]

**Pull request merged**
